sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/Extensions/ServiceResponseExtensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Azure.WebPubSub.AspNetCore
+{
+    /// <summary>
+    /// Help methods for ServiceResponse.
+    /// </summary>
+    public static class ServiceResponseExtensions
+    {
+        /// <summary>
+        /// Set connection states.
+        /// </summary>
+        /// <param name="response"></param>
+        /// <param name="key"></param>
+        /// <param name="value"></param>
+        public static void SetState(this ServiceResponse response, string key, object value)
+        {
+            // value exists.
+            if (response.States.TryGetValue(key, out var value1))
+            {
+                response.States[key] = value;
+            }
+            // value new.
+            else
+            {
+                response.States.Add(key, value);
+            }
+        }
+
+        /// <summary>
+        /// Clear all states.
+        /// </summary>
+        /// <param name="response"></param>
+        public static void ClearStates(this ServiceResponse response)
+        {
+            response.States = null;
```

Comment:
Is the design that `States` can never be repopulated after calling `ClearStates()`?
Otherwise, the setter has to ensure that the dictionay is initialized.

You could do `response.States.Clear();`

Reply:
The state object in ServiceResponse is different from the one in ConnectionContext. Set null here is to distinguish user's operation of clear all states(null) and no change(default dictionary).

So right, I need to check states null in the SetState method.